pkg/apis/triggers/v1alpha1/trigger_types.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TriggerSpec represents a connection between TriggerBinding, Params,
+// and TriggerTemplate; TriggerBinding provides extracted values for
+// TriggerTemplate to then create resources from.
+type TriggerSpec struct {
+	Bindings []*TriggerBinding `json:"bindings"`
+	Template TriggerTemplate   `json:"template"`
+	// +optional
+	Name         string                `json:"name,omitempty"`
+	Interceptors []*TriggerInterceptor `json:"interceptors,omitempty"`
```

Comment:
I don't think Interceptor should be like this. Keep like it's there in EventListener Spec Triggers.

Reply:
After discussing with the team, we will move the structure definition from event_listener_type.go to this file itself.